master/reference/node/configuration.md

```diff
@@ -27,3 +29,54 @@ The `calico/node` container is primarily configured through environment variable
 | ETCD_KEY_FILE     | Path to the etcd key file (optional)       | /etc/calico/key.pem | |
 | ETCD_CERT_FILE    | Path to the etcd client cert (optional)    | /etc/calico/cert.pem | |
 | ETCD_CA_CERT_FILE | Path to the etcd CA file (optional)        | /etc/calico/ca.pem | |
+
+
+### IP Autodetection methods
+
+The following describe the available IP autodetection methods.
+
+#### first-found
+The `first-found` option enumerates all interface IP addresses and returns the
+first valid IP address (best guess based on IP version and type of address) on 
+the first valid interface.  Certain known "local" interfaces
+are omitted, such  as the docker bridge.  Note that order that interfaces are 
+listed is system dependent.
+
+This is the default detection method. However, since this method only makes a
+very simplified guess, it is recommended to either configure the node with a 
+specific IP address, or to use one of the other detection methods.
+
+e.g.
+```
+IP_AUTODETECT_METHOD=first-found
+IP6_AUTODETECT_METHOD=first-found
+```
+
+#### can-reach=DESTINATION
+The `can-reach` method uses your local routing to determine which IP address 
+will be used to reach the supplied destination.  Both IP addresses and domain
+names may be used.
+
+e.g.
+```
+# Using IP addresses
+IP_AUTODETECT_METHOD=can-reach=8.8.8.8
+IP6_AUTODETECT_METHOD=can-reach=2001:4860:4860::8888
+
+# Using domain names
+IP_AUTODETECT_METHOD=can-reach=www.google.com
+IP6_AUTODETECT_METHOD=can-reach=www.google.com
+```
+
+#### interface=INTERFACE-REGEX
+The `interface` method uses the supplied interface regular expression (golang
+syntax) to enumerate matching interfaces and to return the first IP address on
+the first matching interface.  Note that order that interfaces are listed is
+system dependent.
+
+e.g.
+```
+# Valid IP address on interface eth0
+IP_AUTODETECT_METHOD=interface=eth0
+IP6_AUTODETECT_METHOD=interface=eth0
```

Comment:
can we maybe add an example regex here? So users know if they need to put quotes or wherever around the regex while passing it to the variable